charmcraft.yaml

```diff
@@ -0,0 +1,176 @@
+    type: oci-image
+    description: OCI image for Vault
+    upstream-source: ghcr.io/canonical/vault:1.16.3
+
+storage:
+  vault-raft:
+    type: filesystem
+    minimum-size: 10G
+  config:
+    type: filesystem
+    minimum-size: 5M
+  certs:
+    type: filesystem
+    minimum-size: 5M
+  tmp:
+    type: filesystem
+    minimum-size: 5G
+
+peers:
+  vault-peers:
+    interface: vault-peer
+
+provides:
+  vault-autounseal-provides:
+    interface: vault-autounseal
+  vault-kv:
+    interface: vault-kv
+  vault-pki:
+    interface: tls-certificates
+  metrics-endpoint:
+    interface: prometheus_scrape
+  send-ca-cert:
+    interface: certificate_transfer
+    description: |
+      Send our CA certificate so clients can trust the CA by means of forming a relation.
+  grafana-dashboard:
+    interface: grafana_dashboard
+
+assumes:
+  - juju >= 3.1.8
+  - k8s-api
+
+requires:
+  vault-autounseal-requires:
+    interface: vault-autounseal
+    limit: 1
+  ingress:
+    interface: ingress
+    limit: 1
+  tls-certificates-access:
+    interface: tls-certificates
+    limit: 1
+    description: |
+      Communication between the vault units and from a client to Vault should 
+      be done using the certificates provided by this integration.
+  tls-certificates-pki:
+    interface: tls-certificates
+    limit: 1
+    description: |
+      Interface to be used to provide Vault with its CA certificate. Vault will
+      use this certificate to sign the certificates it issues on the `vault-pki` interface.
+  logging:
+    interface: loki_push_api
+  s3-parameters:
+    interface: s3
+
+type: "charm"
+bases:
+  - build-on:
+    - name: "ubuntu"
+      channel: "22.04"
+    run-on:
+    - name: "ubuntu"
+      channel: "22.04"
+
+parts:
+  charm:
+    build-packages:
+      - cargo
+      - libffi-dev
+      - libssl-dev
+      - pkg-config
+      - rustc
+
+actions:
+  authorize-charm:
+    description: >-
+      Authorizes the charm to be able to interact with Vault to manage its
+      operations. A token is required for Vault to use to create the app role and
+      the policy the charm will use to interact with Vault. This token must be
+      placed in juju secret, and this secret should be granted to the charm.
+    params:
+      secret-id:
+        type: string
+        description: >-
+          A secret id from juju that contains a token for Vault that can create 
+          new policies, such as the root token that is provided upon initializing 
+          Vault. Used to create the app role and policy for the charm. It is not
+          stored by the charm.
+    required: [secret-id] 
+
+  create-backup:
+    description: >-
+      Creates a snapshot of the Raft backend and saves it to the S3 storage.
+      Returns backup ID.
+
+  list-backups:
+    description: >-
+      Lists all available backups.
+
+  restore-backup:
+    description: >-
+      Restores a backup from the S3 storage to the Raft backend.
+      Returns backup ID.
```

Comment:
s/Returns backup ID/Returns the backup ID

Reply:
fixed in #499